common/lib/api-client/index.js

```diff
@@ -1,38 +1,56 @@
 const JsonApi = require('devour-client')
 
-const { API, FILE_UPLOADS } = require('../../../config')
+const { API, FILE_UPLOADS, FEATURE_FLAGS } = require('../../../config')
 
 const {
   auth,
+  cacheKey,
   errors,
+  getCache,
+  gotErrors,
+  gotRequest,
+  gotRequestTransformer,
+  gotResponse,
   post,
   request,
-  cacheKey,
-  getCache,
   requestHeaders,
   requestInclude,
   requestTimeout,
 } = require('./middleware')
 const models = require('./models')
 
+let requestMiddleware = request
+let requestMiddlewareName = 'app-request'
+
+if (FEATURE_FLAGS.GOT) {
+  requestMiddleware = gotRequest
+  requestMiddlewareName = 'got-request'
+}
+
 module.exports = function (req) {
   const instance = new JsonApi({
     apiUrl: API.BASE_URL,
     logger: false,
   })
 
-  instance.replaceMiddleware('errors', errors)
+  instance.replaceMiddleware('errors', FEATURE_FLAGS.GOT ? gotErrors : errors)
+
+  if (FEATURE_FLAGS.GOT) {
+    instance.replaceMiddleware('response', gotResponse)
+  }
+
   instance.replaceMiddleware('POST', post(FILE_UPLOADS.MAX_FILE_SIZE))
   instance.replaceMiddleware(
     'axios-request',
-    request({
+    requestMiddleware({
       cacheExpiry: API.CACHE_EXPIRY,
       useRedisCache: API.USE_REDIS_CACHE,
+      timeout: API.TIMEOUT,
```

Comment:
if the request middleware isn't got-request, would this timeout still be needed/work as expected?

Reply:
Good spot. It's not strictly *needed* by the original request middleware, however, because the options are passed in an object and only the required options are destructured it doesn't have any adverse effects to pass in this extra option to both.

I could have wrapped the two calls in an `if/else` statement but that would also require doing that statement every time we instantiate an instance of the API client. This felt the lesser of the two evils and as (I'm hoping) its only temporary I left it like this.